src/unicode.cpp

```diff
@@ -618,7 +618,13 @@ std::vector<uint32_t> unicode_cpts_from_utf8(const std::string & utf8) {
     result.reserve(utf8.size());
     size_t offset = 0;
     while (offset < utf8.size()) {
-        result.push_back(unicode_cpt_from_utf8(utf8, offset));
+        try {
+            result.push_back(unicode_cpt_from_utf8(utf8, offset));
+        }
+        catch (const std::invalid_argument & /*ex*/) {
+            ++offset;
+            result.emplace_back(0xFFFD); // replacement character
+        }
```

Comment:
Please add a comment explaining this logic. This is a hack and we will need to deal with this properly at some point.

Reply:
Done.